ext/node/polyfills/internal/buffer.mjs

```diff
@@ -1192,6 +1233,35 @@ function _hexSlice(buf, start, end) {
   return out;
 }
 
+function adjustOffset(offset, length) {
+  // Use Math.trunc() to convert offset to an integer value that can be larger
+  // than an Int32. Hence, don't use offset | 0 or similar techniques.
+  offset = MathTrunc(offset);
+  if (offset === 0) {
+    return 0;
+  }
+  if (offset < 0) {
+    offset += length;
+    return offset > 0 ? offset : 0;
+  }
+  if (offset < length) {
+    return offset;
+  }
+  return NumberIsNaN(offset) ? 0 : length;
+}
+
+Buffer.prototype.subarray = function subarray(start, end) {
+  const srcLength = this.length;
+  start = adjustOffset(start, srcLength);
+  end = end !== undefined ? adjustOffset(end, srcLength) : srcLength;
+  const newLength = end > start ? end - start : 0;
+  return new FastBuffer(
+    TypedArrayPrototypeGetBuffer(this),
+    TypedArrayPrototypeGetByteOffset(this) + start,
+    newLength,
+  );
+};
+
 Buffer.prototype.slice = function slice(start, end) {
   return this.subarray(start, end);
```

Comment:
This seems wrong. Slice is meant to return a copy, not a new view, right?

Reply:
Yeah I thought so too at first, but [apparently it’s right](https://nodejs.org/api/buffer.html#bufslicestart-end)